lightning/src/ln/channelmanager.rs

```diff
@@ -2985,6 +2987,7 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 								network_update: None,
 								all_paths_failed,
 								path: path.clone(),
+								short_channel_id: Some(path.first().unwrap().short_channel_id),
```

Comment:
I'm not really a huge fan of "punishing" our own channels - on one hand its nice that the payment goes through, but on the other hand it seems like we're addressing the problem in the wrong way - we have a peer who isn't able to read our HTLC-add messages, we shouldn't just stop sending payments their way, we should close the channel with them and disconnect.

Reply:
That would make sense, I suppose.